### ISSUE TITLE
OCI purls are missing repository_url

### ISSUE THREAD

**Issue by jonjohnsonjr** [chainguard-dev/apko#410]
Is it intentional that the OCI purls have no `repository_url`?

```
$ crane blob cgr.dev/chainguard/static@sha256:0f1cad9a9b51c0eb49d0f91f72640541e4b9a976510e73a807fd6bb7bc36aa33 | jq .packages[0].externalRefs[0].referenceLocator -r
pkg:oci/static@sha256:87c802c6011260141ecd54425a2ee498f4f8260ad77ce423a364aaf1a5840a27?mediaType=application%2Fvnd.oci.image.index.v1%2Bjson
```

**Comment by jonjohnsonjr** [comment]
cc @puerco

**Comment by puerco** [comment]
Ah yes, this is intentional. When writing this, @mattmoor and me decided not to add the repository URL to make the purls portable when copying images.

**Comment by mattmoor** [comment]
Yeah, think of these as "relative" purls since things like images of an index must be colocated anyhow.

In contexts like base image (for `ko`), we include them because they aren't necessarily colocated.

**Comment by jonjohnsonjr** [comment]
Makes sense -- it seems safe to assume that absent a repository_url, the image should be in the same repository as the SBOM?

**Comment by coreydaley-cg** [comment]
The OCI purl now carries `repository_url` when the image repository is known. Closing.
